crypto/property/property.c

```diff
@@ -64,6 +64,7 @@ struct ossl_method_store_st {
     OSSL_LIB_CTX *ctx;
     SPARSE_ARRAY_OF(ALGORITHM) *algs;
     CRYPTO_RWLOCK *lock;
+    CRYPTO_RWLOCK *biglock;
```

Comment:
Could you please add some comment(s) here as for what is the purpose of each of the locks?

Reply:
Yes, IMO comments about what the lock protects should be required.
Also document any lock ordering considerations with other locks (e.g., when both are locked, biglock must be locked first and unlocked last) and scenarios where there are fields that can be read with either lock held but both must be held for writing.
I think https://github.com/freebsd/freebsd-src/blob/c4a6d412c9c75fb50603f1ae52d6bd78f8510e6c/sys/sys/vnode.h#L78-L176 is a great example of clear locking documentation, and https://github.com/torvalds/linux/blob/feb9c5e19e913b53cb536a7aa7c9f20107bb51ec/Documentation/filesystems/directory-locking.rst covers some situations where interactions between multiple locks must be considered.

I also am a strong proponent for adding as many assertions about lock status as you can (even if compile-time conditional under something other than `NDEBUG`).

Reply:
Also see below in the OpenSSL code where we have documented the locking requirements:

https://github.com/openssl/openssl/blob/cb2764f2a8165421dc5ab52159af99cbf766fa2c/crypto/provider_core.c#L56-L116

Reply:
I will get to it, thanks for pointing this out